src/index.ts

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { MapboxDraw } from "@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw";
+
+export class MapBoxPropTypes {
+  boxSelect = PropTypes.bool;
+  clickBuffer = PropTypes.number;
+  controls = PropTypes.shape({
+    point: PropTypes.bool,
+    line_string: PropTypes.bool,
+    polygon: PropTypes.bool,
+    trash: PropTypes.bool,
+    combine_features: PropTypes.bool,
+    uncombine_features: PropTypes.bool
+    });
+  default_mode = PropTypes.string;
+  displayControlsDefault = PropTypes.bool;
+  keybindings = PropTypes.bool;
+  modes = PropTypes.object;
+  position = PropTypes.string;
+  onDrawActionable = PropTypes.func;
+  onDrawCombine = PropTypes.func;
+  onDrawCreate = PropTypes.func;
+  onDrawDelete = PropTypes.func;
+  onDrawModeChange = PropTypes.func;
+  onDrawRender = PropTypes.func;
+  onDrawSelectionChange = PropTypes.func;
+  onDrawUncombine = PropTypes.func;
+  onDrawUpdate = PropTypes.func;
+  touchBuffer = PropTypes.number;
+  touchEnabled = PropTypes.bool;
+
+  styles = PropTypes.arrayOf(PropTypes.object);
+}
+
+export default class DrawControl extends React.Component {
+  static contextTypes = {
```

Comment:
Idem here, no need for contextTypes. Create a `interface Context { map }`, and the type of the map field is `import { Map } from 'mapbox-gl'`;